backend/config/config_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestNewDefaultAppConfigLanguage(t *testing.T) {
+	tt := []struct{ locale, wantLang string }{
+		{"", "en"},
+		{"-", "en"},
+		{"en", "en"},
+		{"en-US", "en-US"},
+		{"en_US", "en-US"},
+		{"en-something", "en"},
+		{"de", "de"},
+		{"de-DE", "de-DE"},
+		{"de-CH", "de-CH"},
+	}
+	for _, test := range tt {
+		appconf := NewDefaultAppConfig(test.locale)
+		if v := appconf.Frontend["userLanguage"].(string); v != test.wantLang {
+			t.Errorf("%s: userLanguage=%q; want %s", test.locale, v, test.wantLang)
+		}
+	}
+}
+
+func TestNewConfigTemplate(t *testing.T) {
+	template := func() AppConfig {
+		return AppConfig{
+			Frontend: map[string]interface{}{"userLanguage": "fr"},
+		}
+	}
+
+	// Empty app config file.
+	confFile, err := ioutil.TempFile("", "appconfig1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(confFile.Name()) //nolint:errcheck
+	confFile.Close()                 //nolint:errcheck
+	conf, err := NewConfig(confFile.Name(), "", template)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if v := conf.appConfig.Frontend["userLanguage"].(string); v != "fr" {
+		t.Errorf("userLanguage = %q; want 'fr'", v)
+	}
+
+	// Manually set lang in app config.
+	confFile2, err := ioutil.TempFile("", "appconfig2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(confFile2.Name()) //nolint:errcheck
```

Comment:
maybe the linter could be disabled for tests somehow

Reply:
done in https://github.com/digitalbitbox/bitbox-wallet-app/pull/922